### DISCUSSION TITLE
Questions about Extpos with Optitrack

### DISCUSSION THREAD

**Discussion by yakuman2800** [bitcraze/discussions#1007] [Q&A]
I bought a CrazyFlie 2.1 and am currently attempting to get it to hover.
I have Optitrack set up and passing position information (x,y,z) using the send_extpos method, then trying to get it to reach a position setpoint.

I'm using a modified version of the ramp_motors method provided in [](url)https://github.com/bitcraze/crazyflie-lib-python/blob/c1c42f7a8b58bc95d660247d5a2a6832755dae90/examples/motors/ramp.py#L82

[rampmotors_code.txt](https://github.com/bitcraze/discussions/files/13198995/rampmotors_code.txt)

I attached a sample of what my code looks like.
When I launch my script, the drone immediately flips.

My suspicion is that I am not using the send_extpose method, which includes a quaternion as input, and I plan on implementing this shortly.

But I am noticing that some of your examples like [](url)https://www.bitcraze.io/2017/04/updated-autonomous-flight-using-opencv-and-kalman-filter/ don't seem to pass angle information to the drone.

Is angle information required from an external source to allow the CrazyFlie to go to position setpoints, or is the onboard gyroscope sufficient?

Furthermore, how do I check that extpos info is successfully sent to the CrazyFlie? I have the Pose Logger setup to read pose info from the CrazyFlie, and it looks like a fusion between my mocap system and the CrazyFlie's internal sensors, but I wonder if there's a more conclusive method.

Thank you!

**Comment by knmcguire** [comment]
Hi! Before going any further, it seems that perhaps you should first check out this new example for your optitrack system: https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/mocap/mocap_hl_commander.py

Also, you want to have the crazyflie follow setpoints but you are using the ramp motor example, which runs the motors without any control. You need to send setpoints such that the internal controllers on the crazyflie can do something with it. Please check out the documentation about the stabilization module: https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/sensor-to-control/. Also check out the step by step guide for the crazyflie library: https://www.bitcraze.io/documentation/tutorials/#software

About checking if the crazyflie gets the right positioning, there is a function that wait until the position converges before taking of. https://github.com/bitcraze/crazyflie-lib-python/blob/c1c42f7a8b58bc95d660247d5a2a6832755dae90/examples/mocap/mocap_hl_commander.py#L108. If you are not using any other deck for positioning (like the flowdeck) this is a pretty good indication if the crazyflie is getting a position

**Reply by yakuman2800** [reply]
I'm trying to run [](url)https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/mocap/mocap_hl_commander.py but I can't import motioncapture. When running "pip install motioncapture -vvv" it says that it's skipping the link for most of the files.

**Reply by yakuman2800** [reply]
I also tried to install:
- the previous version of motioncapture
- the latest version of motioncapture manually by downloading the files
and in both cases, I got a CMake Error.

**Reply by krichardsson** [reply]
Are you on python 3.12 perhaps?
There are a lot of packets that are not compiled for 3.12 yet and is this typically the type of problem I see on new python versions. 
I use python 3.11 and this works

**Reply by yakuman2800** [reply]
I've tried python 3.10.6 and 3.11.5 and in both cases a massive amount of links are skipped, so only the dist-info folder is created.

**Reply by krichardsson** [reply]
It sounds like some sort of python/pip problem but I'm not sure what is going on. Do other pip installs work?
@whoenig do you have any input?
